Add .flutterflowignore support

tmpsantos opened the pull request:
This file allows for users to create a list of files to be ignored when exporting the project code (works a bit like .gitignore).

Files matching the globbing rules listed at .flutterflowignore are going to be ignored. This might break the project if changes in these files are needed to compile the project, so it should be used with extreme caution.

This functionality was previously achieved with:
https://github.com/krabhishek/flutterflow-filtered-pull

tmpsantos commented:
cc @aemelyanovff

aemelyanovff commented:
> cc @aemelyanovff

I'll review it hopefully by the end of the week. I'm a bit swamped right now, sorry.

aemelyanovff commented on lib/src/flutterflow_ignore.dart:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  bool matches(String path) {
+    for (final glob in _globs) {
+      if (glob.matches(path)) {
+        return true;
+      }
+    }
+
+    return false;
```
Nit: `return _globs.any((glob) => glob.matches(path));` is a little more idiomatic in Dart.

tmpsantos replied:
Done.

aemelyanovff commented on lib/src/flutterflow_api_client.dart:
```diff
@@ -135,25 +133,28 @@ Future<String?> exportCode({
+
+      // Found on .flutterflowignore, move on.
+      if (ignore.matches(relativeFilename)) {
+        print('Ignoring $relativeFilename, file remained unchanged.');
```
Nit: logs like this usually go to `stderr` (it's debatable of course).

tmpsantos replied:
Makes sense, the rest of the code is like this.

aemelyanovff commented on test/fixtures/.flutterflowignore:
```diff
@@ -0,0 +1,8 @@
+file
+dir/*
```
When I add `dir/` to `.gitignore` it ignores the entire directory, but this implementation doesn't. Let's update it to support ignoring directories?

tmpsantos replied:
Done, also added support for comments starting with `#` like in git.

aemelyanovff reviewed (commented):
Very nice!
I have one suggestion and a couple of nits.

tmpsantos commented:
Thanks @aemelyanovff for the review. Addressed your concerns and added more tests and comments support. Also made the path being match always relative to where the `.flutterflowignore` is located for the sake of consistence.

aemelyanovff commented:
LGTM!
